web/contexts/serverConfig.tsx

```diff
@@ -16,10 +20,66 @@ type Props = {
   children: ReactNode;
 };
 
+type InnerProps = Props & {
+  children: ReactNode;
+  data: ServerConfigContextValue | undefined;
+  error: any;
+};
+
+// Custom hook to fetch the server config using IPC
+function useServerConfigIPC(): {
+  data: ServerConfigContextValue | undefined;
+  error: any;
+} {
```

Comment:
Can you move this return type to a declared type? Can also use the type for `InnerProps` since they share `data` and `error`